Update datatables-fixedheader

maruilian11 opened the pull request:
Pull request for issue: #8742 
@cdnjs/library-reviewer Please help me review this PR, thanks.
**I not only update js files of this lib but also add css according to the author's suggestion.**
https://github.com/DataTables/FixedHeader/issues/83#issuecomment-243998230

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/DataTables/FixedHeader
 * Official website (optional, not the repository): https://datatables.net/extensions/fixedheader/
 * NPM package url (optional): https://www.npmjs.com/package/datatables.net-fixedheader
 * License and its reference: MIT,
https://github.com/DataTables/Dist-DataTables-FixedHeader/blob/master/License.txt
https://github.com/DataTables/Dist-DataTables-FixedHeader-DataTables/blob/master/License.txt
https://github.com/DataTables/Dist-DataTables-FixedHeader-Bootstrap/blob/master/License.txt
https://github.com/DataTables/Dist-DataTables-FixedHeader-Foundation/blob/master/License.txt
https://github.com/DataTables/Dist-DataTables-FixedHeader-jQueryUI/blob/master/License.txt

 * GitHub / Bitbucket popularity (required):
   Watch 8
   Star 50
   Fork 68
 * NPM download stats (optional): 
   12 downloads in the last day
   767 downloads in the last week
   3,039 downloads in the last month

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo 
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

pvnr0082t commented on ajax/libs/datatables-fixedheader/package.json:
```diff
@@ -19,5 +19,17 @@
+      {
+        "basePath": "js",
+        "files": [
+          "**/*"
```
It's a little bit not safe or no sense to use **/* under a non-dist folder. We can use `dataTables.fixedHeader*` here.

pvnr0082t commented on ajax/libs/datatables.net-fixedheader-dt/package.json:
```diff
@@ -0,0 +1,35 @@
+      {
+        "basePath": "css",
+        "files": [
+          "**/*"
```
It's a little bit not safe or no sense to use **/* under a non-dist folder. We can use `fixedHeader.dataTables*` here.

pvnr0082t commented on ajax/libs/datatables.net-fixedheader-bs/package.json:
```diff
@@ -0,0 +1,36 @@
+      {
+        "basePath": "css",
+        "files": [
+          "**/*"
```
It's a little bit not safe or no sense to use **/* under a non-dist folder. We can use `fixedHeader.bootstrap*` here.

pvnr0082t commented on ajax/libs/datatables.net-fixedheader-zf/package.json:
```diff
@@ -0,0 +1,36 @@
+      {
+        "basePath": "css",
+        "files": [
+          "**/*"
```
It's a little bit not safe or no sense to use **/* under a non-dist folder. We can use `fixedHeader.foundation*` here.

pvnr0082t commented on ajax/libs/datatables.net-fixedheader-jqui/package.json:
```diff
@@ -0,0 +1,36 @@
+      {
+        "basePath": "css",
+        "files": [
+          "**/*"
```
It's a little bit not safe or no sense to use **/* under a non-dist folder. We can use `fixedHeader.jqueryui*` here.

maruilian11 commented:
@pvnr0082t Because they are dist repos, you can tell it by the repo name, I think we can use `**/*` as the filemap?
I will modify commit msg later.

sufuf3 commented:
Agree with @pvnr0082t 
Because it's not under `dist` or `build` ... etc.
But under `js` and `css` which is the same with `src`
I think it's a good suggestion.
@maruilian11 What do you think?

Thank you.

maruilian11 commented:
@PeterDaveHello Can we use `**/*` in dist repo?

PeterDaveHello commented:
> Can we use **/* in dist repo?

Yes we can.

maruilian11 commented:
@cdnjs/library-reviewer @sufuf3 Please help me continue reviewing this PR, thank you.

sufuf3 commented:
@maruilian11 we can use `**/*` in `dist` of basePath, but we can't use `**/*` in `css` or `js` of basePath.
I think need to modify these, such as from A to B:
- A
```
    "fileMap": [
      {
        "basePath": "css",
        "files": [
          "**/*"
        ]
      }
    ]
```
- B
```
    "fileMap": [
      {
        "basePath": "",
        "files": [
          "css/*"
        ]
      }
    ]
```

cc @PeterDaveHello

PeterDaveHello commented:
In dist repo, it depends, I think you can discuss. Anyway, we prefer not using `css` or `js` as basePath, also, there is conflict needs to be fixed first.

maruilian11 commented:
@PeterDaveHello @sufuf3  But if we don't set `css` and `js` as basepath, `datatables-fixedheader`'s file structure will be different from the existed versions.
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/datatables-fixedheader

So do we only set `css` and `js` as basepath for `datatables-fixedheader`?

sufuf3 commented:
I think we can't delete the files of v2.0.6, v2.1.0, v2.1.1 
But we can add new file structure in v2.0.6, v2.1.0, v2.1.1 
(keep the origin file structure also add new file structure in these old versions)

I've met this situation before but I need to find it later.

Thank you.

PeterDaveHello commented:
@sufuf3 Nice catch!

maruilian11 commented:
@sufuf3 It's ok. I remember that case.

maruilian11 commented:
@cdnjs/library-reviewer Please help me review this PR. Thanks!

maruilian11 commented:
1. I have mentioned the info I updated in the second line of 6d176fe
2. Yes, we can fetch `dataTables.fixedHeader.js` with auto-update config, but the point is we **can't** fetch `dataTables.fixedHeader.min.js`. And we prefer to use the official minified file, so I add them from the website manually.
3. I update the commit to use symbol links. Thanks.
4. I found there are more versions(prefix with RELEASE_*) need to be ignored, so I updated the commit. Please take a look.
5.  Yes. Thanks. I added `v` for them.
6. The reason I use `js/**/*` or `css/**/*` in this PR is they are **dist** repo. You can tell that by looking at their repo name. And I have ask this for Peter before. https://github.com/cdnjs/cdnjs/pull/11532#issuecomment-330526944
7. About the keyword in https://github.com/cdnjs/cdnjs/pull/11532/commits/89f969a9c8027d07dd03772be80e87efbb7d9f71, I just follow the upstream.

sufuf3 commented on .gitignore:
```diff
@@ -231,6 +231,7 @@ node_modules/
 /ajax/libs/angular-gridster/v.0.10.4
 /ajax/libs/numbered/0.0.4
 /ajax/libs/numbered/1.0.0
+/ajax/libs/datatables-fixedheader/RELEASE*/
```
I can't find `RELEASE` prefix, did I miss something?
https://github.com/DataTables/Dist-DataTables-FixedHeader/tags
But maybe it's fine here.
Because you met before.

extend1994 replied:
There is no versions with prefix in `datatables-fixedheader`. I think it would be better to remove https://github.com/cdnjs/cdnjs/pull/11532/commits/4fd5a0fbc265f2eeeffbc22faf41c0624ac91502. Sorry for not noticing last time. Maybe you saw other library like  `datatables` or `datatables-colvis`? Or the author could have deleted tags.

extend1994 commented on ajax/libs/datatables-fixedheader/package.json:
```diff
@@ -1,7 +1,7 @@
-  "version": "2.1.1",
-  "filename": "dataTables.fixedHeader.min.js",
+  "version": "3.1.3",
+  "filename": "js/dataTables.fixedHeader.min.js",
```
Not sure if this change should be in the commit involving "auto-update" update because filename's path is the relative path to `basePath` in auto-update config.

extend1994 commented on ajax/libs/datatables.net-fixedheader-bs/package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "datatables.net-fixedheader-bs",
```
Maybe `DataTables-FixedHeader-Bootstrap` which is the same as the part of git repo name (also auto-update config) would be better and more meaningful?

extend1994 commented on ajax/libs/datatables.net-fixedheader-jqui/package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "datatables.net-fixedheader-jqui",
```
Maybe `DataTables-FixedHeader-jQueryUI` which is the same as the part of git repo name (also auto-update config) would be better and more meaningful?

extend1994 commented on ajax/libs/datatables.net-fixedheader-zf/package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "datatables.net-fixedheader-zf",
```
Maybe `DataTables-FixedHeader-Foundation` which is the same as the part of git repo name (also auto-update config) would be better and more meaningful?

extend1994 commented on ajax/libs/datatables.net-fixedheader-zf/package.json:
```diff
@@ -0,0 +1,36 @@
+    "table",
+    "Foundation"
+  ],
+  "homepage": "https://datatables.net/extensions/fixedheader",
```
Would README.md in git repo be better? I can't find info about this styling library in the current link.

extend1994 commented on ajax/libs/datatables.net-fixedheader-bs/package.json:
```diff
@@ -0,0 +1,36 @@
+    "table",
+    "Bootstrap"
+  ],
+  "homepage": "https://datatables.net/extensions/fixedheader",
```
Would README.md in git repo be better? I can't find info about this styling library in the current link.

extend1994 commented on ajax/libs/datatables.net-fixedheader-dt/package.json:
```diff
@@ -0,0 +1,35 @@
+    "jQuery",
+    "table"
+  ],
+  "homepage": "https://datatables.net/extensions/fixedheader",
```
Would README.md in git repo be better? I can't find info about this styling library in the current link.
